spec/importers/file_importers/courses_sections_importer_spec.rb

```diff
@@ -31,8 +33,63 @@
         it 'creates a section of the course' do
           expect(Section.first.course).to eq(Course.first)
         end
+
+        it 'logs one row processed, one row created' do
+          expected_statuses = [
+            "processed": 1,
+            "excluded": 0,
+            "created": 1,
+            "updated": 0,
+            "deleted": 0,
+            "rejected": 0
+          ]
+          expect(import_record_detail.rows_summary).to match_array(expected_statuses)
+        end
       end
 
+      context 'updates correctly' do
+        let!(:section) {FactoryGirl.create(:section)}
+        let(:row) { { course_number:section.course_number,
+                      course_description:section.course_description,
+                      section_number:section.section_number,
+                      school_local_id:section.course.school.local_id,
+                      term_local_id:section.term_local_id,
+                      section_schedule:section.schedule,
+                      section_room_number:'NEW ROOM'
+                  } }
+        let!(:import_record_detail) {FactoryGirl.create(:import_record_detail)}
+
+          before do
+            importer = described_class.new(nil, nil, log, nil, import_record_detail)
+            importer.import_row(row)
+          end
+
+          it 'leaves the course' do
+            expect(Course.count).to eq(1)
+          end
+
+          it 'leaves the section' do
+            expect(Section.count).to eq(1)
+          end
+
+          it 'updates the section' do
+            section.reload
+            expect(section.room_number).to eq("NEW ROOM")
+          end
+
+          it 'logs one row processed, one row updated' do
+            expected_statuses = [
```

Comment:
Interesting data structure, kind of an array/hash combination? Can you say a little more about this?

When I run it in the console I get:

```
2.3.0 :001 >           expected_statuses = [
2.3.0 :002 >                 "processed": 1,
2.3.0 :003 >                 "excluded": 0,
2.3.0 :004 >                 "created": 0,
2.3.0 :005 >                 "updated": 0,
2.3.0 :006 >                 "deleted": 0,
2.3.0 :007 >                 "rejected": 1
2.3.0 :008?>             ]
 => [{:processed=>1, :excluded=>0, :created=>0, :updated=>0, :deleted=>0, :rejected=>1}] 
2.3.0 :009 > 
```

So it looks like it becomes an array with a hash inside. Would it be simpler to just use a hash? If not, should we explicitly show the hash within the array to make it clear what the data structure is?

Reply:
Yeah, sorry got my Ruby datastructures a bit mixed up. ;-)